### PR TITLE
#144 specific on_change

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**

### CODE REVIEW

File: gui/doc/on_change.csv [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿Name,Type,Default,Doc
+on_change,str,,"The name of a function that is triggered when the value is updated.
+The parameters of that function are all optional:
+<ul>
+<li>gui instance</li>
```
Comment (FabienLelaquais) [inline gui/doc/on_change.csv]: gui?

File: gui/doc/on_change.csv [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿Name,Type,Default,Doc
```
Comment (FabienLelaquais) [inline gui/doc/on_change.csv:1]: I guess we'll have to fill this in...

File: taipy/gui/gui.py [hunk 3]
```diff
@@ -269,27 +271,30 @@ def _update_var(self, var_name: str, value: t.Any, propagate=True, holder: _Taip
...
-                        if isinstance(value, _MapDict)
-                        else value
-                    )
-                self.on_change(*args)
+                    args[2] = value
+                on_change_fn(*args)
             except Exception as e:
-                warnings.warn(f"on_change: function invocation exception: {e}")
-        self.__send_var_list_update(list(modified_vars), var_name)
+                warnings.warn(f"{on_change or 'on_change'}: function invocation exception: {e}")
```
Comment (FabienLelaquais) [inline taipy/gui/gui.py]: Hmmm.
May be "'on_change' callback function raised an exception: {e}"?

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
I havent't seen the 'date' control factory update...
Nor did I see the 'slider' documentation update.

Shouldn't 'expandable' be an on_action component?

File: gui/doc/on_change.csv (hunk 1 above)
Comment (FredLL-Avaiga) [reply gui/doc/on_change.csv]: fixed

File: gui/doc/on_change.csv (hunk 2 above)
Comment (FredLL-Avaiga) [reply gui/doc/on_change.csv:1]: What do you mean ?

**Comment by FredLL-Avaiga** [comment]
> I havent't seen the 'date' control factory update... 

It's in there

> Nor did I see the 'slider' documentation update.

Fixed
 
> Shouldn't 'expandable' be an on_action component?

why ?

File: gui/doc/on_change.csv (hunk 2 above)
Comment (FabienLelaquais) [reply gui/doc/on_change.csv:1]: I mean we need doc on the parameters. No big deal.

**Review by FabienLelaquais — APPROVED** [review]
Excellentissimo

**Review by FabienLelaquais — APPROVED** [review]
Looks perfect to me!

**Review by dinhlongviolin1 — APPROVED** [review]
Great job

**Pull request merged**